subsys/bluetooth/mesh/adv.h

```diff
@@ -47,9 +48,10 @@ struct bt_mesh_adv {
 extern const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES];
 
 /* xmit_count: Number of retransmissions, i.e. 0 == 1 transmission */
-struct net_buf *bt_mesh_adv_create(enum bt_mesh_adv_type type,
-				   enum bt_mesh_adv_tag tag,
-				   uint8_t xmit, k_timeout_t timeout);
+struct net_buf *bt_mesh_adv_main_create(enum bt_mesh_adv_type type,
+					uint8_t xmit, k_timeout_t timeout);
+
+struct net_buf *bt_mesh_adv_relay_create(bool hp, uint8_t xmit);
```

Comment:
It will be nice to have a linked list sorted according to priority and more granularity for priorities. The problem with inefficient usage of two FIFOs will go away. Priority will be always actual. All available memory will be used in an efficient manner. Customers can customize priorities for different frames.

Reply:
it's a good idea

Reply:
My concern is that the need to dynamically adjust the list each time may lead to cpu overhead

Reply:
Currently, two pool method, always to ensure high priority messages have buffer to relay, instead of all being consumed by low-priority messages.

Reply:
Yes, maybe we can remove from list, the lowest priority messages will be free for higher priority messages.

Reply:
This frequent traversal of the list needs to be thread-safe.

Reply:
Although sysworkq cannot be interrupted, application tasks may be preempted, so race conditions may occur

Reply:
Could `SYS_SLIST_FOR_EACH_CONTAINER_SAFE` help to avoid race conditions?

Reply:
No, I used mutex for thread protection.

Reply:
I also don't delete nodes in the list traversal. I'm worried about different tasks accessing the queue, so it is possible to lock through mutex.

Reply:
![image](https://user-images.githubusercontent.com/26378606/201298498-6e483bd4-6d52-4204-bcf0-708f4dbd3898.png)

All tests great, maybe we need add new testcase for this priority relay.

But it's better to do this after you two have finished the review. I don't want to do it in rebase.

Reply:
One important problem found.

net_buf_unref will destroy the nodes behind the queue.